lib/workers/repository/update/branch/index.ts

```diff
@@ -553,26 +553,29 @@ export async function processBranch(
     await setStability(config);
     await setConfidence(config);
 
-    // break if we pushed a new commit because status check are pretty sure pending but maybe not reported yet
+    // new commit means status check are pretty sure pending but maybe not reported yet
+    // if PR has not been created + new commit + prCreation !== immediate skip
     // but do not break when there are artifact errors
     if (
-      !config.artifactErrors?.length &&
-      !userRebaseRequested &&
       commitSha &&
-      config.prCreation !== 'immediate'
+      config.prCreation !== 'immediate' &&
+      !branchPr &&
+      !config.artifactErrors?.length &&
+      !userRebaseRequested
```

Comment:
the check here is for `!userRebaseRequested` with `commitSha` though.
it could be a commitSha for update or create, it doesn't have to be rebase.

but the good question is, why is it in here?
so i think it's there because it's with `config.prCreation !== 'immediate'` 
so if there's a pr creation that should wait, and there's a rebase request, then we shouldnt wait, we should just continue, 
so i think the check is OK, do u agree with me? or you still think we should remove it?